src/lib/types.ts

```diff
@@ -209,14 +210,319 @@ type ResolveInviteCodeResponse = {
  * similar words for this), we've created or changed our application data
  * which is further defined below.
  */
-type ApplicationEvent = CalendarCreatedEvent;
 
-type CalendarCreatedEvent = {
-  type: "calendar_created";
+type ApplicationEvent =
+  | UserNameAssigned
+  | CalendarCreated
+  | CalendarUpdated
+  | CalendarDeleted
+  | CalendarAccessRequested
+  | CalendarAccessAccepted
+  | CalendarAccessRejected
+  | CalendarAccessRejected
+  | SpaceCreated
+  | SpaceUpdated
+  | SpaceDeleted
+  | ResourceCreated
+  | ResourceUpdated
+  | ResourceDeleted
+  | EventCreated
+  | EventUpdated
+  | EventDeleted
+  | SpaceRequested
+  | SpaceRequestAccepted
+  | SpaceRequestRejected
+  | SpaceRequestAcceptanceRevoked
+  | ResourceRequested
+  | ResourceRequestAccepted
+  | ResourceRequestRejected
+  | ResourceRequestAcceptanceRevoked
+  | UserRoleAssigned;
+
+/**
+ * Reused types
+ */
+
+type SpaceRequestId = Hash;
+type ResourceRequestId = Hash;
+
+type CalendarFields = {
+  calendarName: string;
+  calendarDates: TimeSpan[];
+};
+
+type SpaceFields = {
+  type: "physical" | "gps" | "virtual";
+  name: string;
+  location: PhysicalLocation | GPSLocation | VirtualLocation;
+  capacity: number;
+  accessibility: string;
+  description: string;
+  contact: string;
+  message: string;
+  link: Link;
+  images: Image[];
+  availability: TimeSpan[] | "always";
+  multiBookable: boolean;
+};
+
+type ResourceFields = {
+  name: string;
+  ownerId: PublicKey;
+  description: string;
+  contact: string;
+  link: Link;
+  images: Image[];
+  availability: TimeSpan[] | "always";
+  multiBookable: boolean; // resource can be booked more than once in the same time span
+};
+
+type EventFields = {
+  name: string;
+  description: string;
+  location?: SpaceRequestId; // ref to a space
+  startDate: Date; // allocated time of a space
+  endDate: Date; // allocated time of a space
+  publicStartDate?: Date; // public facing
+  publicEndDate?: Date; // public facing
+  resources: ResourceRequestId[];
+  links: Link[];
+  images: Image[];
+};
+
+/**
+ * User
+ *
+ * NOTE: sent on `inbox` channel
+ */
+
+type UserNameAssigned = {
+  type: "user_name_assigned";
```

Comment:
Maybe this is too concrete, why not `profile_updated` and keep the option open for more fields in the future (profile picture, etc.)?